btfgen.c

```diff
@@ -274,56 +344,123 @@ static struct btf_reloc_type *btf_reloc_put_type(struct btf *btf,
 		break;
 	/* tells if some other type needs to be handled */
 	default:
-		printf("unsupported relocation: %d\n", reloc_type->id);
+		printf("unsupported kind: %s (%d)\n", btf_kind_str(btf_kind(reloc_type->type)), reloc_type->id);
 		return ERR_PTR(-EINVAL);
 	}
 
 	return reloc_type;
 }
 
-/* Return pointer to btf_reloc_type by id */
-static struct btf_reloc_type *btf_reloc_get_type(struct btf_reloc_info *info, int id) {
-	struct btf_reloc_type *type = NULL;
+/*
+ * Same as btf_reloc_put_type, but adding all fields, from given complex type, recursively
+ */
+static struct btf_reloc_type *btf_reloc_put_type_all(struct btf *btf,
```

Comment:
What do you think about making this function return an integer? I don't think it makes sense to return a pointer as we are adding more than one type, also the returned pointer is not used anywhere.

Reply:
Yep, I confess I thought about it at some point. Will do this change.